File: PP_29-01-25/BuiltInFuntion/Temperature.cs
using System;

public class Temperature
{
    // Method to convert Fahrenheit to Celsius
    public static double FahrenheitToCelsius(double fahrenheit)
    {
        return (fahrenheit - 32) * 5 / 9;
    }

    // Method to convert Celsius to Fahrenheit
    public static double CelsiusToFahrenheit(double celsius)
    {
        return (celsius * 9 / 5) + 32;
    }

    // main method
    public static void Main(string[] args)
    {
        //input for conversion type
        Console.WriteLine("Select conversion type:");
        Console.WriteLine("1. Fahrenheit to Celsius");
        Console.WriteLine("2. Celsius to Fahrenheit");
        Console.Write("Enter your choice (1 or 2): ");

        int choice = Convert.ToInt32(Console.ReadLine());

        if (choice == 1)
        {
            Console.Write("Enter temperature in Fahrenheit: ");
            double fahrenheit = Convert.ToDouble(Console.ReadLine());
            double celsius = FahrenheitToCelsius(fahrenheit);
            Console.WriteLine(fahrenheit + "°F is equal to " + celsius + "°C");
        }
        else if (choice == 2)
        {
            Console.Write("Enter temperature in Celsius: ");
            double celsius = Convert.ToDouble(Console.ReadLine());
            double fahrenheit = CelsiusToFahrenheit(celsius);
            Console.WriteLine(celsius + "°C is equal to " + fahrenheit + "°F");
        }
        else
        {
            Console.WriteLine("Invalid choice.");
        }
    }
}

File: PP_17-03-25/Function/temperature.js
const prompt = require("prompt-sync")();

function convertTemperature(choice, temp) {
    switch (choice) {
        case '1': // Celsius to Fahrenheit
            if (temp < 0 || temp > 100) {
                console.log("Temperature out of range (0°C - 100°C).");
                return;
            }
            console.log(`${temp}°C = ${(temp * 9/5) + 32}°F`);
            break;

        case '2': // Fahrenheit to Celsius
            if (temp < 32 || temp > 212) {
                console.log("Temperature out of range (32°F - 212°F).");
                return;
            }
            console.log(`${temp}°F = ${((temp - 32) * 5/9).toFixed(2)}°C`);
            break;

        default:
            console.log("Invalid choice.");
    }
}

// User Input
console.log("1: Convert Celsius to Fahrenheit\n2: Convert Fahrenheit to Celsius");
let choice = prompt("Enter your choice (1/2): ");
let temp = parseFloat(prompt("Enter temperature: "));

convertTemperature(choice, temp);
